### PR TITLE
Use quotes when formatting table names

### PR DESCRIPTION
**Pull request by buremba**
Table names may contain special characters and should be quoted when formatting SQL queries. For example, when we try to create a view using a query similar to `CREATE VIEW test AS SELECT count(*) FROM "$test"`, Presto fails because it tries to format the query.

### CODE REVIEW

**Comment by electrum** [comment]
Use formatName()

**Comment by electrum** [comment]
Can you add a test for this? In `TestStatementBuilder`, there is this line:
```java
printStatement("select \"TOTALPRICE\" \"my price\" from \"ORDERS\"");
```
Just change it to this (so the table name has a space):
```java
printStatement("select \"TOTALPRICE\" \"my price\" from \"MY ORDERS\"");
```

**Comment by electrum** [comment]
Looks like some of the tests are failing with this change. Please see the Travis build failures.

**Comment by buremba** [comment]
@electrum it fails to create docs because of this error: `PyException: GC overhead limit exceeded`

File: presto-parser/src/main/java/com/facebook/presto/sql/SqlFormatter.java [hunk 1]
```diff
@@ -327,7 +328,11 @@ protected Void visitAllColumns(AllColumns node, Integer context)
         @Override
         protected Void visitTable(Table node, Integer indent)
         {
-            builder.append(node.getName().toString());
+            builder.append(node.getName().getOriginalParts()
+                    .stream()
```
Comment (electrum) [inline presto-parser/src/main/java/com/facebook/presto/sql/SqlFormatter.java]: Nit: we usually put `.stream()` on the previous line

File: presto-parser/src/main/java/com/facebook/presto/sql/SqlFormatter.java [hunk 2]
```diff
@@ -327,7 +328,11 @@ protected Void visitAllColumns(AllColumns node, Integer context)
...
+                    .map(Formatter::formatName)
+                    .collect(Collectors.joining(".")));
```
Comment (electrum) [inline presto-parser/src/main/java/com/facebook/presto/sql/SqlFormatter.java]: Nit: `joining()` is a good method to static import

**Comment by electrum** [comment]
Please wrap the "description" of Git commit messages to ~70 characters when it's normal text (for stuff like a benchmark output which is pre-formatted text it can be as wide as necessary).

File: presto-parser/src/main/java/com/facebook/presto/sql/SqlFormatter.java [hunk 3]
```diff
@@ -327,7 +328,11 @@ protected Void visitAllColumns(AllColumns node, Integer context)
         @Override
         protected Void visitTable(Table node, Integer indent)
         {
-            builder.append(node.getName().toString());
+            builder.append(node.getName().getOriginalParts()
```
Comment (electrum) [inline presto-parser/src/main/java/com/facebook/presto/sql/SqlFormatter.java]: This can use the `formatName()` function that takes a `QualifiedName`.

**Comment by electrum** [comment]
Merged, thanks!

**Pull request closed without merging**